### PR TITLE
Additional Text Relating to Performance

### PR DESCRIPTION
**Pull request by jamesmcclain**
Still Needs
- [x] Discussion of Mutable Types
- [x] ~~Discussion of Primitive Types (pasted-in from Nathan's writing?)~~ Tile Heirarchy
- [x] Disucssion of Spark

### CODE REVIEW

File: docs/architecture/high-performance-scala.md [hunk 1]
```diff
@@ -1,8 +1,18 @@
 # JVM Primitives
 ------
```
Comment (fosskers) [inline docs/architecture/high-performance-scala.md]: The `#` and `---` combination creates some funky double-underlining when rendered by github. Either `##` or `#` alone for these should work.

**Comment by fosskers** [comment]
You could also talk about the `Tile` type hierarchy that we use, to avoid boxing.

File: docs/architecture/high-performance-scala.md [hunk 2]
```diff
@@ -90,9 +100,45 @@ readability. They've been introduced merely to overcome shortcomings in
...
 
+## Loops ##
 
-# Macros
------
-# Spark
-------
\ No newline at end of file
+A commonly-seen feature throughout the codebase is the use of `cfor`- or `while`-loops where it would seem that an ordinary `for`-loop would be sufficient.
+We avoid using "`for`-loops" in most cases because they are more than just loops in Scala.
```
Comment (pomadchin) [inline docs/architecture/high-performance-scala.md]: Dunno how to correct to name `for` a sa `loop`; as it is a `for comprehension`, that's why it is slow; http://docs.scala-lang.org/tutorials/FAQ/yield.html

Comment (jamesmcclain) [inline docs/architecture/high-performance-scala.md]: Yeah, I was trying to decide how to approach it, as well, that is why I put it in quotation marks.

**Comment by jamesmcclain** [comment]
> You could also talk about the Tile type hierarchy that we use to avoid boxing.

I am hoping that @moradology 's `CellType` discussion would covered most of that (though I have not seen it).  Can anyone confirm or deny?

**Comment by fosskers** [comment]
It does a bit, yeah.

**Comment by fosskers** [comment]
I just confirmed; mention of the tile hierarchy and why we use it isn't mentioned in the Cell Type explanation.

**Comment by jamesmcclain** [comment]
Thanks for all of the useful feedback ... incorporating.

**Comment by pomadchin** [comment]
Can we add information about `TileCombiners`, and to mention that we generate some boilerplate: https://github.com/geotrellis/geotrellis/blob/master/project/Boilerplate.scala ?

**Comment by jamesmcclain** [comment]
> Can we add information about TileCombiners, and to mention that we generate some boilerplate: https://github.com/geotrellis/geotrellis/blob/master/project/Boilerplate.scala ?

Certainly, thanks for the input.  I will add that when there is time.

**Comment by jamesmcclain** [comment]
I believe that all prior comments have been addressed.

**Pull request merged**